app/src/main/kotlin/com/mapzen/erasermap/util/NotificationBroadcastReceiver.kt

```diff
@@ -0,0 +1,32 @@
+package com.mapzen.erasermap.util
+
+import android.app.Activity
+import android.app.NotificationManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.mapzen.erasermap.view.MainActivity
+
+public class NotificationBroadcastReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        if (intent.getBooleanExtra(NotificationCreator.EXIT_NAVIGATION, false)) {
+            cancelAllNotifications(context)
+            startBaseActivityWithExitExtra(context)
+        }
+    }
+
+    private fun cancelAllNotifications(context: Context) {
+        val notificationManager = context.getSystemService(Activity.NOTIFICATION_SERVICE) as NotificationManager
+        notificationManager.cancelAll()
+    }
+
+    private fun startBaseActivityWithExitExtra(context: Context) {
+        var exitRoutingIntent: Intent? = Intent(context, MainActivity::class.java)
```

Comment:
Local value `exitRoutingIntent` can be declared immutable with `val` rather than `var`. Also no need to declare as nullable so type can be inferred.

    val exitRoutingIntent = Intent(context, MainActivity::class.java)

This will allow us to access via `.` below instead of `?.`

Also line ending semicolons can be dropped.